### PR TITLE
"Add client SSL cert+key" step now expects string or readable stream

### LINKED ISSUE #39: Cannot use parseType file in step /^I secure the connection with certificate (.+) and with key (.+)$/
The parser that is called for type `file` creates a readable stream, it does not return the file contents as a String, unlike for example type `testdata`, which reads a testdata file using `readfileSync` and parses it as JSON.

`parseValueString('file randomfile.txt');`

This creates a problem in the step to add a client cert and key to the request, because that step expects a string, not a file stream.

Step:
`/^I secure the connection with certificate (.+) and with key (.+)$/ `

### PR DESCRIPTION
**Pull request by FrankZZ**
This PR fixes #39

### CODE REVIEW

**Comment by lucamaraschi** [comment]
Update the package.json and then I will merge the pull request

File: lib/steps/http-steps.js [hunk 1]
```diff
@@ -43,18 +42,60 @@ module.exports = function httpSteps() {
         };
     }
 
+    function isReadableStream(obj) {
+        return typeof (obj._read === 'function') &&
```
Comment (gabrieleds) [inline lib/steps/http-steps.js]: Use a different condition:
```
return (typeof obj._read === 'function') && (typeof obj._readableState === 'object');
```
'cause
```
(obj._read === 'function') // => true
typeof true // => "boolean"
```

File: lib/steps/http-steps.js [hunk 2]
```diff
@@ -43,18 +42,60 @@ module.exports = function httpSteps() {
...
+            typeof (obj._readableState === 'object');
+    }
+
+    function consumeReadStreamOrString (stream, callback) {
+        try {
```
Comment (gabrieleds) [inline lib/steps/http-steps.js]: The `try` is not needed

File: lib/steps/http-steps.js [hunk 3]
```diff
@@ -43,18 +42,60 @@ module.exports = function httpSteps() {
...
+            if (isReadableStream(stream)) {
+                var data = [];
+                stream.on('data', function (chunk) {
+                    data.push(chunk);
+                });
+                stream.on('end', function () {
+                    return callback(null, data.join(''));
```
Comment (gabrieleds) [inline lib/steps/http-steps.js]: `return` is not needed

File: lib/steps/http-steps.js [hunk 4]
```diff
@@ -43,18 +42,60 @@ module.exports = function httpSteps() {
...
+                });
+            } else if (typeof stream === 'string' ||
+                Object.toString.call(stream) === '[object String]') {
+                    return callback(null, stream);
```
Comment (gabrieleds) [inline lib/steps/http-steps.js]: `return` is not needed

File: lib/steps/http-steps.js [hunk 5]
```diff
@@ -43,18 +42,60 @@ module.exports = function httpSteps() {
...
+            } else {
+                return callback(new Error('Cannot consume this object.'));
```
Comment (gabrieleds) [inline lib/steps/http-steps.js]: Same here

File: lib/steps/http-steps.js (hunk 1 above)
Comment (gabrieleds) [inline lib/steps/http-steps.js]: or use [is-readable-stream](https://github.com/julien-f/nodejs-is-readable-stream/blob/master/index.js)

File: lib/steps/http-steps.js (hunk 2 above)
Comment (lucamaraschi) [inline lib/steps/http-steps.js]: But it's missing the on.('error') handler for the stream

File: lib/steps/http-steps.js (hunk 1 above)
Comment (lucamaraschi) [inline lib/steps/http-steps.js]: I would then go for [is-readable-stream](https://github.com/julien-f/nodejs-is-readable-stream/blob/master/index.js)

**Pull request merged**